Sources/Swift/Integrations/SessionReplay/SentrySessionReplay.swift

```diff
@@ -93,12 +96,29 @@ class SentrySessionReplay: NSObject {
         delegate?.sessionReplayStarted(replayId: sessionReplayId)
     }
 
+    func pause() {
+        isSessionPaused = true
+        videoSegmentStart = nil
```

Comment:
`m`: I'm a bit confused about why we need a lock for `videoSegmentStart` in `resume` and here we don't. Is that intended?

Reply:
the lock in `resume` is for the `displayLink` not the `videosegmentStart`